owncloudApp/src/main/java/com/owncloud/android/ui/preview/PreviewImageFragment.kt

```diff
@@ -0,0 +1,389 @@
+            }
+            R.id.action_unset_available_offline -> {
+                mContainerActivity.fileOperationsHelper.toggleAvailableOffline(file, false)
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun seeDetails() {
+        mContainerActivity.showDetails(file)
+    }
+
+    override fun onDestroy() {
+        if (bitmap != null) {
+            bitmap.recycle()
+            System.gc()
+            // putting this in onStop() is just the same; the fragment is always destroyed by
+            // {@link FragmentStatePagerAdapter} when the fragment in swiped further than the
+            // valid offscreen distance, and onStop() is never called before than that
+        }
+        super.onDestroy()
+    }
+
+    /**
+     * Opens the previewed image with an external application.
+     */
+    private fun openFile() {
+        mContainerActivity.fileOperationsHelper.openFile(file)
+        finish()
+    }
+
+    override fun onTransferServiceConnected() {
+        if (progressController != null) {
+            progressController!!.startListeningProgressFor(file, account)
+        }
+    }
+
+    override fun onFileMetadataChanged(updatedFile: OCFile) {
+        file = updatedFile
+        requireActivity().invalidateOptionsMenu()
+    }
+
+    override fun onFileMetadataChanged() {
+        val storageManager = mContainerActivity.storageManager
+        if (storageManager != null) {
+            file = storageManager.getFileByPath(file.remotePath)
+        }
+        requireActivity().invalidateOptionsMenu()
+    }
+
+    override fun onFileContentChanged() {
+        loadAndShowImage()
+    }
+
+    override fun updateViewForSyncInProgress() {
+        progressController!!.showProgressBar()
+    }
+
+    override fun updateViewForSyncOff() {
+        progressController!!.hideProgressBar()
+    }
+
+    private fun loadAndShowImage() {
+        val requestOptions =
+            RequestOptions.diskCacheStrategyOf(DiskCacheStrategy.NONE).apply(RequestOptions.skipMemoryCacheOf(true))
+        Glide.with(requireContext())
+            .load(File(file.storagePath))
+            .apply(requestOptions)
+            .listener(object : RequestListener<Drawable?> {
+                override fun onLoadFailed(
+                    e: GlideException?, model: Any, target: Target<Drawable?>,
+                    isFirstResource: Boolean
+                ): Boolean {
+                    Timber.e(e, "Error loading image")
+                    return false
+                }
+
+                override fun onResourceReady(
+                    resource: Drawable?, model: Any, target: Target<Drawable?>,
+                    dataSource: DataSource, isFirstResource: Boolean
+                ): Boolean {
+                    Timber.d("Loading image %s", file.fileName)
+                    return false
+                }
+            })
+            .into(photo_view!!)
```

Comment:
When we transform some java code to Kotlin, we should try to improve it. There are a lot of `!!` in this file that could be improved. It would be beneficial in the long-term from my point of view. What do you think?

Reply:
Ok, I removed them. But it's always a question where to start and where to end. 
I only want to have it done